php_phongo.c

```diff
@@ -354,35 +354,35 @@ void phongo_writeconcern_init(zval *return_value, const mongoc_write_concern_t *
 zend_bool phongo_writeconcernerror_init(zval *return_value, bson_t *bson TSRMLS_DC) /* {{{ */
 {
 	bson_iter_t iter;
-	php_phongo_writeconcernerror_t *writeconcernerror;
+	php_phongo_writeconcernerror_t *intern;
 
-	writeconcernerror = Z_WRITECONCERNERROR_OBJ_P(return_value);
+	object_init_ex(return_value, php_phongo_writeconcernerror_ce);
+
+	intern = Z_WRITECONCERNERROR_OBJ_P(return_value);
 
 	if (bson_iter_init_find(&iter, bson, "code") && BSON_ITER_HOLDS_INT32(&iter)) {
-		writeconcernerror->code = bson_iter_int32(&iter);
+		intern->code = bson_iter_int32(&iter);
 	}
+
 	if (bson_iter_init_find(&iter, bson, "errmsg") && BSON_ITER_HOLDS_UTF8(&iter)) {
 		uint32_t errmsg_len;
 		const char *err_msg = bson_iter_utf8(&iter, &errmsg_len);
 
-		writeconcernerror->message = estrndup(err_msg, errmsg_len);
+		intern->message = estrndup(err_msg, errmsg_len);
 	}
+
 	if (bson_iter_init_find(&iter, bson, "errInfo") && BSON_ITER_HOLDS_DOCUMENT(&iter)) {
-		uint32_t               len;
-		const uint8_t         *data;
+		uint32_t       len;
+		const uint8_t *data = NULL;
 
 		bson_iter_document(&iter, &len, &data);
 
-		if (!data) {
-			return false;
-		}
```

Comment:
With `BSON_ITER_HOLDS_DOCUMENT()` guarding this block, `bson_iter_document()` will always populate the document's data. This removes an odd case where we might have returned false without throwing. If the data is somehow null after this (I can't imagine how), `phongo_bson_to_zval()` will catch and throw an error when initializing the BSON reader.